sirepo/package_data/static/js/sirepo-components.js

```diff
@@ -1871,6 +1871,66 @@ SIREPO.app.directive('validatedString', function(panelState, validationService)
     };
 });
 
+SIREPO.app.directive('viewLogIframe', function() {
+    return {
+        restrict: 'A',
+        scope: {
+            hideLink: '<',
+            logPath: '<',
+            wrapperViewLog: '<',
+            wrapperDownloadLog: '<'
+        },
+        template: `
+            <a href data-ng-click="viewLog()" id="viewLogIframeLink">{{ linkText }}</a>
```

Comment:
It took me a minute to figure out the logic for the view log wrappers. It may be worth setting up a meeting and we can go over it together. But, my basic thought is that this line belongs in the wrappers and not in this directive. If you do that then I think some of the tricky bits of code with setting the scan id and needing a timeout to let a digest cycle happen will go away. Instead we can use bootsrap.show and bootstrap `shown.bs.modal` (if you search you will see other uses of this) to show the modal and then call a callback when it is shown.  Likewise, I think we'll be able to get rid of the hideLink boolean. That should live in the wrappers (or maybe we just change it to "show log" everywhere) but still  I think moving this line into the wrappers will eliminate the need for hiding in the raydata case.

Reply:
Is this close to what you had in mind? 27f9e3765b906a03ddb4fa4099ea53ebb2d0272d